### PR TITLE
Support latest package:analyzer

### PR DESCRIPTION
**Pull request by kevmoo**
Prepare to publish several packages

### CODE REVIEW

File: build/pubspec.yaml [hunk 1]
```diff
@@ -21,3 +21,9 @@ dev_dependencies:
   build_test: ^2.0.0
   lints: '>=1.0.0 <3.0.0'
   test: ^1.16.0
+
+dependency_overrides:
```
Comment (jakemac53) [inline build/pubspec.yaml:25]: Do you need these?

**Review by jakemac53 — APPROVED** [review]
LGTM but I think you should be able to remove the overrides?

File: build/pubspec.yaml (hunk 1 above)
Comment (kevmoo) [reply build/pubspec.yaml:25]: I need them for latest analyzer to resolve for tests. I was going to drop when I publish.

**Comment by kevmoo** [comment]
> LGTM but I think you should be able to remove the overrides?

I was going to before I publish. Should I land the PR w/out the overrides?

**Comment by jakemac53** [comment]
> I was going to before I publish. Should I land the PR w/out the overrides?

If they aren't needed I would yes - I realize now you probably did this to allow you to get the latest analyzer? But we can see its green so I would now revert and then land.

**Comment by jakemac53** [comment]
Oh actually - it looks like you still did not get the latest analyzer (at least for some of the runs).

You might want to add an override to the latest analyzer to force it.

**Comment by jakemac53** [comment]
Scratch my last comment - looks like the dev builds did pick it up

File: build_resolvers/pubspec.yaml [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+version: 2.0.10
 description: Resolve Dart code in a Builder
 repository: https://github.com/dart-lang/build/tree/master/build_resolvers
 
 environment:
   sdk: ">=2.17.0 <3.0.0"
 
 dependencies:
-  analyzer: ^4.6.0
+  analyzer: '>=4.4.0 <6.0.0'
```
Comment (natebosch) [inline build_resolvers/pubspec.yaml:10]: Is this safe to downgrade the min version?

```suggestion
  analyzer: '>=4.6.0 <6.0.0'
```

File: build_runner_core/pubspec.yaml [hunk 3]
```diff
@@ -27,7 +27,7 @@ dependencies:
   yaml: ^3.0.0
 
 dev_dependencies:
-  analyzer: ^4.6.0
+  analyzer: '>=4.4.0 <6.0.0'
```
Comment (natebosch) [inline build_runner_core/pubspec.yaml:30]:
```suggestion
  analyzer: '>=4.6.0 <6.0.0'
```

File: build_test/pubspec.yaml [hunk 4]
```diff
@@ -24,6 +24,6 @@ dependencies:
   watcher: ^1.0.0
 
 dev_dependencies:
-  analyzer: ^4.6.0
+  analyzer: '>=4.4.0 <6.0.0'
```
Comment (natebosch) [inline build_test/pubspec.yaml:27]:
```suggestion
  analyzer: '>=4.6.0 <6.0.0'
```

**Review by natebosch — APPROVED** [review]

**Comment by kevmoo** [comment]
Thanks @natebosch – copy/paste error

**Pull request merged**